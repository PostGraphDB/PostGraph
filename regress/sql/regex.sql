/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

LOAD 'postgraph';
SET search_path TO postgraph;

CREATE GRAPH regex;
USE GRAPH regex;

CREATE (n:Person {name: 'John'}) ;
CREATE (n:Person {name: 'Jeff'}) ;
CREATE (n:Person {name: 'Joan'}) ;

--
-- =~ cypher case sensitive regular expression 
--
MATCH (n:Person) WHERE n.name =~ 'JoHn' RETURN n;
MATCH (n:Person) WHERE n.name =~ '(?i)JoHn' RETURN n;
MATCH (n:Person) WHERE n.name =~ 'Jo.n' RETURN n;
MATCH (n:Person) WHERE n.name =~ 'J.*' RETURN n;

--
-- ~ POSIX case sensitive regular expression 
--
MATCH (n:Person) WHERE n.name ~ 'JoHn' RETURN n;
MATCH (n:Person) WHERE n.name ~ '(?i)JoHn' RETURN n;
MATCH (n:Person) WHERE n.name ~ 'Jo.n' RETURN n;
MATCH (n:Person) WHERE n.name ~ 'J.*' RETURN n;

--
-- ~* POSIX case insensitive regular expression 
--
MATCH (n:Person) WHERE n.name ~* 'john' RETURN n;
MATCH (n:Person) WHERE n.name ~* '(?i)john' RETURN n;
MATCH (n:Person) WHERE n.name ~* 'jo.n' RETURN n;
MATCH (n:Person) WHERE n.name ~* 'j.*' RETURN n;
MATCH (n:Person) WHERE n.name ~* '.*' RETURN n;
MATCH (n:Person) WHERE n.name ~* 'JOHN' RETURN n;

--
-- !~ POSIX case sensitive regular expression not match
--
MATCH (n:Person) WHERE n.name !~ 'john' RETURN n;
MATCH (n:Person) WHERE n.name !~ '(?i)john' RETURN n;
MATCH (n:Person) WHERE n.name !~ 'jo.n' RETURN n;
MATCH (n:Person) WHERE n.name !~ 'j.*' RETURN n;
MATCH (n:Person) WHERE n.name !~ '.*' RETURN n;
MATCH (n:Person) WHERE n.name !~ 'JOHN' RETURN n;

--
-- !~* POSIX case sensitive regular expression not match
--
MATCH (n:Person) WHERE n.name !~* 'john' RETURN n;
MATCH (n:Person) WHERE n.name !~* '(?i)john' RETURN n;
MATCH (n:Person) WHERE n.name !~* 'jo.n' RETURN n;
MATCH (n:Person) WHERE n.name !~* 'j.*' RETURN n;
MATCH (n:Person) WHERE n.name !~* '.*' RETURN n;
MATCH (n:Person) WHERE n.name !~* 'JOHN' RETURN n;

DROP GRAPH regex CASCADE;
