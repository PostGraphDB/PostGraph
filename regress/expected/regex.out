/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
LOAD 'postgraph';
SET search_path TO postgraph;
CREATE GRAPH regex;
NOTICE:  graph "regex" has been created
 create_graph 
--------------
 
(1 row)

USE GRAPH regex;
 use_graph 
-----------
 
(1 row)

CREATE (n:Person {name: 'John'}) ;
--
(0 rows)

CREATE (n:Person {name: 'Jeff'}) ;
--
(0 rows)

CREATE (n:Person {name: 'Joan'}) ;
--
(0 rows)

--
-- =~ cypher case sensitive regular expression 
--
MATCH (n:Person) WHERE n.name =~ 'JoHn' RETURN n;
 n 
---
(0 rows)

MATCH (n:Person) WHERE n.name =~ '(?i)JoHn' RETURN n;
                                     n                                      
----------------------------------------------------------------------------
 {"id": 844424930131969, "label": "person", "properties": {"name": "John"}}
(1 row)

MATCH (n:Person) WHERE n.name =~ 'Jo.n' RETURN n;
                                     n                                      
----------------------------------------------------------------------------
 {"id": 844424930131969, "label": "person", "properties": {"name": "John"}}
 {"id": 844424930131971, "label": "person", "properties": {"name": "Joan"}}
(2 rows)

MATCH (n:Person) WHERE n.name =~ 'J.*' RETURN n;
                                     n                                      
----------------------------------------------------------------------------
 {"id": 844424930131969, "label": "person", "properties": {"name": "John"}}
 {"id": 844424930131970, "label": "person", "properties": {"name": "Jeff"}}
 {"id": 844424930131971, "label": "person", "properties": {"name": "Joan"}}
(3 rows)

--
-- ~ POSIX case sensitive regular expression 
--
MATCH (n:Person) WHERE n.name ~ 'JoHn' RETURN n;
 n 
---
(0 rows)

MATCH (n:Person) WHERE n.name ~ '(?i)JoHn' RETURN n;
                                     n                                      
----------------------------------------------------------------------------
 {"id": 844424930131969, "label": "person", "properties": {"name": "John"}}
(1 row)

MATCH (n:Person) WHERE n.name ~ 'Jo.n' RETURN n;
                                     n                                      
----------------------------------------------------------------------------
 {"id": 844424930131969, "label": "person", "properties": {"name": "John"}}
 {"id": 844424930131971, "label": "person", "properties": {"name": "Joan"}}
(2 rows)

MATCH (n:Person) WHERE n.name ~ 'J.*' RETURN n;
                                     n                                      
----------------------------------------------------------------------------
 {"id": 844424930131969, "label": "person", "properties": {"name": "John"}}
 {"id": 844424930131970, "label": "person", "properties": {"name": "Jeff"}}
 {"id": 844424930131971, "label": "person", "properties": {"name": "Joan"}}
(3 rows)

--
-- ~* POSIX case insensitive regular expression 
--
MATCH (n:Person) WHERE n.name ~* 'john' RETURN n;
                                     n                                      
----------------------------------------------------------------------------
 {"id": 844424930131969, "label": "person", "properties": {"name": "John"}}
(1 row)

MATCH (n:Person) WHERE n.name ~* '(?i)john' RETURN n;
                                     n                                      
----------------------------------------------------------------------------
 {"id": 844424930131969, "label": "person", "properties": {"name": "John"}}
(1 row)

MATCH (n:Person) WHERE n.name ~* 'jo.n' RETURN n;
                                     n                                      
----------------------------------------------------------------------------
 {"id": 844424930131969, "label": "person", "properties": {"name": "John"}}
 {"id": 844424930131971, "label": "person", "properties": {"name": "Joan"}}
(2 rows)

MATCH (n:Person) WHERE n.name ~* 'j.*' RETURN n;
                                     n                                      
----------------------------------------------------------------------------
 {"id": 844424930131969, "label": "person", "properties": {"name": "John"}}
 {"id": 844424930131970, "label": "person", "properties": {"name": "Jeff"}}
 {"id": 844424930131971, "label": "person", "properties": {"name": "Joan"}}
(3 rows)

MATCH (n:Person) WHERE n.name ~* '.*' RETURN n;
                                     n                                      
----------------------------------------------------------------------------
 {"id": 844424930131969, "label": "person", "properties": {"name": "John"}}
 {"id": 844424930131970, "label": "person", "properties": {"name": "Jeff"}}
 {"id": 844424930131971, "label": "person", "properties": {"name": "Joan"}}
(3 rows)

MATCH (n:Person) WHERE n.name ~* 'JOHN' RETURN n;
                                     n                                      
----------------------------------------------------------------------------
 {"id": 844424930131969, "label": "person", "properties": {"name": "John"}}
(1 row)

--
-- !~ POSIX case sensitive regular expression not match
--
MATCH (n:Person) WHERE n.name !~ 'john' RETURN n;
                                     n                                      
----------------------------------------------------------------------------
 {"id": 844424930131969, "label": "person", "properties": {"name": "John"}}
 {"id": 844424930131970, "label": "person", "properties": {"name": "Jeff"}}
 {"id": 844424930131971, "label": "person", "properties": {"name": "Joan"}}
(3 rows)

MATCH (n:Person) WHERE n.name !~ '(?i)john' RETURN n;
                                     n                                      
----------------------------------------------------------------------------
 {"id": 844424930131970, "label": "person", "properties": {"name": "Jeff"}}
 {"id": 844424930131971, "label": "person", "properties": {"name": "Joan"}}
(2 rows)

MATCH (n:Person) WHERE n.name !~ 'jo.n' RETURN n;
                                     n                                      
----------------------------------------------------------------------------
 {"id": 844424930131969, "label": "person", "properties": {"name": "John"}}
 {"id": 844424930131970, "label": "person", "properties": {"name": "Jeff"}}
 {"id": 844424930131971, "label": "person", "properties": {"name": "Joan"}}
(3 rows)

MATCH (n:Person) WHERE n.name !~ 'j.*' RETURN n;
                                     n                                      
----------------------------------------------------------------------------
 {"id": 844424930131969, "label": "person", "properties": {"name": "John"}}
 {"id": 844424930131970, "label": "person", "properties": {"name": "Jeff"}}
 {"id": 844424930131971, "label": "person", "properties": {"name": "Joan"}}
(3 rows)

MATCH (n:Person) WHERE n.name !~ '.*' RETURN n;
 n 
---
(0 rows)

MATCH (n:Person) WHERE n.name !~ 'JOHN' RETURN n;
                                     n                                      
----------------------------------------------------------------------------
 {"id": 844424930131969, "label": "person", "properties": {"name": "John"}}
 {"id": 844424930131970, "label": "person", "properties": {"name": "Jeff"}}
 {"id": 844424930131971, "label": "person", "properties": {"name": "Joan"}}
(3 rows)

--
-- !~* POSIX case sensitive regular expression not match
--
MATCH (n:Person) WHERE n.name !~* 'john' RETURN n;
                                     n                                      
----------------------------------------------------------------------------
 {"id": 844424930131970, "label": "person", "properties": {"name": "Jeff"}}
 {"id": 844424930131971, "label": "person", "properties": {"name": "Joan"}}
(2 rows)

MATCH (n:Person) WHERE n.name !~* '(?i)john' RETURN n;
                                     n                                      
----------------------------------------------------------------------------
 {"id": 844424930131970, "label": "person", "properties": {"name": "Jeff"}}
 {"id": 844424930131971, "label": "person", "properties": {"name": "Joan"}}
(2 rows)

MATCH (n:Person) WHERE n.name !~* 'jo.n' RETURN n;
                                     n                                      
----------------------------------------------------------------------------
 {"id": 844424930131970, "label": "person", "properties": {"name": "Jeff"}}
(1 row)

MATCH (n:Person) WHERE n.name !~* 'j.*' RETURN n;
 n 
---
(0 rows)

MATCH (n:Person) WHERE n.name !~* '.*' RETURN n;
 n 
---
(0 rows)

MATCH (n:Person) WHERE n.name !~* 'JOHN' RETURN n;
                                     n                                      
----------------------------------------------------------------------------
 {"id": 844424930131970, "label": "person", "properties": {"name": "Jeff"}}
 {"id": 844424930131971, "label": "person", "properties": {"name": "Joan"}}
(2 rows)

DROP GRAPH regex CASCADE;
NOTICE:  drop cascades to 3 other objects
DETAIL:  drop cascades to table regex._ag_label_vertex
drop cascades to table regex._ag_label_edge
drop cascades to table regex.person
NOTICE:  graph "regex" has been dropped
 drop_graph 
------------
 
(1 row)

